### PR TITLE
Configure eslint to follow airbnb style guide for client

### PR DESCRIPTION
**Pull request by vgoh1**
# Description 
See title. Scholly/base will come in another PR because this is already huge.

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 

# Problem
Client code isn't very consistent so we should enforce a style guide.

# Solution description
I've suppressed a lot of rules because fixing all of those in this one PR would take a long time, but I took note which ones were important enough to refactor soon-ish and which ones were low-prio. 

# Tests 
If people could check that the functionality still works and call out anything that broke in the process that would be appreciated!

# Relevant document/ link (if any) 
N/A

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
High

### CODE REVIEW

File: client/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        "node": true
+    },
+    "extends": ["eslint:recommended", "airbnb", "prettier"],
+    "parserOptions": {
+        "ecmaVersion": 12
+    },
+    "rules": {
+        "max-len": "off",
+        "no-underscore-dangle": "off",
+        "camelcase": "off",
```
Comment (aksh0001) [inline client/.eslintrc.json:14]: not a question, just an observation: I'm surprised that we're not actually following camelcase

Comment (vgoh1) [reply client/.eslintrc.json:14]: Actually, I just copied this file from /api so the first few rules got carried over here. With the rule on, there's only 1 case of it - several follow up PRs are probably going to come next PI to go through these rules and fix violations.

**Review by aksh0001 — APPROVED** [review]

**Review by rpra0006 — APPROVED** [review]
Code still runs for me

File: client/src/components/publications/Publications.js [hunk 2]
```diff
@@ -83,7 +84,12 @@ const Publications = () => {
         {loading ? (
           <Spinner animation="border" />
         ) : (
-          <h4>Total of {teamPublications.length} publications</h4>
+          <h4>
+            Total of
+            {teamPublications.length}
+            {' '}
```
Comment (jfan0010) [inline client/src/components/publications/Publications.js:90]: need a space between `line 88` and `line 89`

**Review by jfan0010 — APPROVED** [review]
LGTM

File: client/src/components/publications/Publications.js (hunk 2 above)
Comment (vgoh1) [reply client/src/components/publications/Publications.js:90]: thanks for the catch

**Pull request merged**